### PR TITLE
Websockets: Api keys via authorizers

### PR DESCRIPTION
**Pull request by mskg**
## What did you implement:
Updated some missing pieces for ApiGatewayV2 configuration:
- Add AWS::ApiGatewayV2::Api / ApiKeySelectionExpression (header or lambda)
- Add test cases for AWS::ApiGatewayV2::Api / RouteSelectionExpression
- Add AWS::ApiGatewayV2::Route / ApiKeyRequired

~Also tried to add the link between existing usagePlans and the Websockets (related to #6096). The implementation searches for existing plans (provider.usagePlan) and adds its own stage to the existing list. Not sure if this approach is valid, but that worked or our case here.~

## How did you implement it:
n/a

## How can we verify it:
- Testcases have been updated

## Todos:

_**Note: Run `npm run test-ci` to run all validation checks on proposed changes**_

- [X] Write tests and confirm existing functionality is not broken.  
       **Validate via `npm test`**
- [X] Write documentation
- [X] Ensure there are no lint errors.  
       **Validate via `npm run lint-updated`**  
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
- [X] Ensure introduced changes match Prettier formatting.  
       **Validate via `npm run prettier-check-updated`**  
       _Note: All reported issues can be automatically fixed by running `npm run prettify-updated`_
- [X] Make sure code coverage hasn't dropped
- [X] Provide verification config / commands / resources
- [X] Enable "Allow edits from maintainers" for this PR
- [X] Update the messages below

**_Is this ready for review?:_** Yes  
**_Is it a breaking change?:_** NO

### CODE REVIEW

**Comment by mskg** [comment]
> This looks pretty solid from a code perspective. Could you update the documentation and add a `serverless.yml` to the PR description so that we can test it?

Sure, waited for your first comments. Will do that.

**Comment by pmuens** [comment]
> Sure, waited for your first comments. Will do that.

Thanks @mskg :+1:

Let us know when you're done and we'll do another review 😊

**Comment by pmuens** [comment]
Hey @mskg thanks again for working on this! 💪

Just wanted to check-in with you on this PR and see if you'd consider it done. Seems like you've added the tests and the documentation which is great!

Could you maybe compile a `serverless.yml` so that we can verify and merge it into `master` ASAP? Thanks in advance and let us know if you need any help.

File: lib/plugins/aws/package/compile/events/websockets/lib/usagePlan.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+'use strict';
```
Comment (pmuens) [inline lib/plugins/aws/package/compile/events/websockets/lib/usagePlan.js]: We should add a test for this function.

**Comment by pmuens** [comment]
Here's just another quick update on this PR.

I just remove the usage plan functionality to make this PR simpler. We can introduce such feature in a follow-up PR. I tested it today and apparently it's only working if the `websocket` event is used in combination with a `http` event. The reason being that API keys (and usage plans) are only created in the `apiGateway` plugin implementation and not in the `websockets` plugin implementation.

Here's a list of todos we have to work on to get this PR into a mergeable state:

- [ ] Add logic to compile `apiKeys` from the `provider` section. This should work if only `websocket` events are used as well as if `websocket` and `http` events are used in combination. We might want to move the `apiKeys.js` file from the `apiGateway` plugin into a shared directory where both event sources can access the code (this is also true when compiling usage plans)
- [ ] Add a `serverless.yml` definition in this PR so that we can test this functionality manually
- [ ] Turn the `serverless.yml` test case from above into an integration test

**Comment by medikoo** [comment]
Closing as it's not active, not close to be in mergeable state, and it's possibly hard to take it over by someone else.

Still @mskg if you feel like working on this, do not hesitate, I'll be happy to reopen

**Pull request closed without merging**